### ISSUE TITLE
Unable to install certificate/profile in physical iOS 16

### ISSUE THREAD

**Issue by Esquilli** [ProxymanApp/Proxyman#1293]
### Proxyman version? (Ex. Proxyman 1.4.3)
- 3.6.2 (30620)

### macOS Version? (Ex. mac 10.14)
- macOS Monterrey 12.4

### Steps to reproduce
- Same instructions as the "iOS Setup Guide" for physical devices.

### Expected behavior
- Users should be able to install certificates on a physical device.
- When you go to `proxy.man/ssl` the profile is not downloaded.

### Screenshots (optional)
- N/A

**Comment by NghiaTranUIT** [comment]
Hey, iOS 16 beta is buggy when setting the HTTP Proxy via Wifi.

If you're using:
- iOS 16 Beta 1, 2, 3: Turn ON Turn OFF the HTTP Proxy toggle in Wifi might do the trick (not 100%)
- iOS 16 Beta 4 is broken. No way to fix it.

We gonna file a radar on Apple soon.

### You can confirm by
- Could not access to `chls.pro/ssl` from Charles Proxy
- Set HTTP Proxy IP with a random number, and you're still able to access the Internet -> It means it's a bug from Apple -> The Traffic still doesn't go through the IP Proxy.

### Workaround
- You can try [Atlantis framework](https://github.com/ProxymanApp/atlantis) from Proxymam 👍

**Comment by esquillidd** [comment]
Yeah, I can confirm it is not working with the steps you mentioned. Guess I will have to wait 👍

**Comment by avi2k4** [comment]
Having the same issue on iOS 16 beta 7. Proxying isn't working for PortSwigger or Charles either.

**Comment by NghiaTranUIT** [comment]
@avi2k4 some users report that after forgetting your Wifi hotspot, it can fix the problem 👍

**Comment by phr85** [comment]
I having problems since I use Xocde 14 / iOS 16. I try to see https traffic, but it only stuck at yellow dot (ACTIVE). 

<img width="1680" alt="Screenshot 2022-09-21 at 09 25 29" src="https://user-images.githubusercontent.com/1322091/191400397-c07ed66d-13c5-416e-bea9-fe1aa4013aea.png">

Is this issue one described in this ticket or is mine related to something else?
(I reseted Root Certificate on Mac and reinstall it in Simulator)

**Comment by NghiaTranUIT** [comment]
@phr85  Look like you're using the old Atlantis library version. I remember that I've fixed this issue on iOS 16. Please update the latest version: https://github.com/ProxymanApp/atlantis/releases/tag/1.18.2

**Comment by phr85** [comment]
@NghiaTranUIT Thank you. This solves my issue.

**Comment by haivudang88** [comment]
hi @NghiaTranUIT 
is there any solution to this Bug already? 
i could not download and install Charles proxy for the devices upgrade from iOS 16 Beta (1 to Gold)
Thank you so much

Chào @NghiaTranUIT ,
Bạn có thông tin thêm về lỗi này không? 
Mình không thể tải và sử dụng Charles proxy cho các máy được update từ iOS 16 Beta
Cám ơn bạn nhiều

**Comment by NghiaTranUIT** [comment]
@haivudang88 If you're using the iOS 16 Beta, there is an Apple Bug that affects Proxyman & Charles Proxy app too.

Basically, the iOS 16 beta doesn't respect the HTTP Proxy in the Wifi, you the `proxy.man/ssl` local website is inaccessible.

Here is a document: https://docs.proxyman.io/troubleshooting/ios-16-devices-issues

**Comment by jkapsouras** [comment]
I have 16.2 beta on my device and the problem is still active. proxy.man/ssl is not accessible

**Comment by NghiaTranUIT** [comment]
Me too. The crazy bug since iOS 16.0 and they haven't fixed it yet :/

**Comment by joaodforce** [comment]
iOS 16.2 is out, and it still does not address this.

**Comment by NghiaTranUIT** [comment]
It's sad news, but it's true. 

I've reopened the bug ticket on Apple Radar. Hopefully they will address it soon 👍

**Comment by karthikchalla5** [comment]
@NghiaTranUIT Is it possible to share the link raised with Apple for this bug ? so we can track it from there. It looks like taking more time than expected. and we are blocked from using the tool on iOS 16.x

**Comment by NghiaTranUIT** [comment]
I've opened on Apple Feedback Assistant: https://feedbackassistant.apple.com/feedback/11871219

Not sure if you can access it.

**Comment by vdavydovHH** [comment]
I don't sure will it help
But legitimate proxy works well on iOS 16.x

Problem with proxy only specific to MITM

**Comment by NghiaTranUIT** [comment]
Tested with iOS 16.1, using HTTP Proxy with Manual Mode doesn't work with any IP, e.g. 192.168.1.100, http://proxyman.local, etc

My iPhone is still able to connect to the Internet in spite of invalid Proxy IP. Not sure if it's fixed on iOS 16.2.

**Comment by jkapsouras** [comment]
A (not very nice for me) workaround is to use the mobile app (https://proxyman.io/ios). The mobile app works fine and captures requests and responses directly from the device.

**Comment by NghiaTranUIT** [comment]
yup, it's true because the Proxyman for iOS app is using the VPN (not the HTTP Proxy), so it can capture all data.

Tip: You can share the traffic from the iOS app to the macOS device, and preview it with Proxyman for macOS 👍

**Comment by ivandev7** [comment]
I am testing with Proxyman & Charles on iOS 16.2 and I can capture http/https traffic but websocket connections are not showing and I need this part.

**Comment by NghiaTranUIT** [comment]
@ivandev7 It's a known issue from URLSessionWebsocketTask. WS traffic doesn't go through the HTTP Proxy. Thus, it never appears on the Proxy app.

To capture it, please use [Atltantis framework](https://github.com/ProxymanApp/atlantis) from Proxyman. Basically, it captures WS/WSS traffic by using method swizzling, and send to the macOS app to preview 👍

**Comment by jkapsouras** [comment]
Just install 16.3 to my device and the proxy works. I think problem is solved from apple.

**Comment by NghiaTranUIT** [comment]
Wow, it's a great news 🚀🚀🚀

I will upgrade to iOS 16.3 and let other people know 👋

**Comment by pathikb** [comment]
Sadly doesn't work for me on iOS 16.3!

**Comment by NghiaTranUIT** [comment]
Bad news: Apple hasn't fixed the HTTP Proxy bug on iOS 16.3 🐛 

Tried to set the HTTP Proxy to random IP/Port, but it still connect to the Internet ❌ 😢

**Comment by jasonhibbs** [comment]
I was able to get Proxyman working today by forgetting and reconnecting to the network I want to use. 
I was having problems on 16.3 before this.

**Comment by tiagobnobrega** [comment]
Forgetting and reconnecting didn't work for me. I noticed that if I connect to the router without an internet connection (remove the cable from the router) proxy will work. Of course, this setup does not work for me as I still need an internet connection. Curious how having an internet connection affects this

**Comment by NghiaTranUIT** [comment]
@tiagobnobrega it's definitely an Apple bug : < I've opened it on Apple Feedback is still not resolved by Apple Team.

**Comment by aaadonai** [comment]
I am still experiencing this issue on iOS 16.4.1. Anyone else still having issues?

**Comment by NghiaTranUIT** [comment]
It's fixed on my iOS 16.4.1 devices. @aaadonai You should forget your Wifi, reconnect it again, restart the device also fix it.

**Comment by ratelgogo** [comment]
> It's fixed on my iOS 16.4.1 devices. @aaadonai You should forget your Wifi, reconnect it again, restart the device also fix it.

ios 16.4.1, forget Wifi, reconnect it again,  alse restart the device , it does not work :cry

**Comment by pinkhominid** [comment]
Still no luck with iPhone X and iPhone 13 both running 16.4.1 (a) after suggested workarounds.

**Comment by NghiaTranUIT** [comment]
@pinkhominid I'm not sure how Apple fixes it. On my devices, iPhone 13 Proxy Max iOS 16.4.1 (a), and some users, It's fixed after forgetting the Wifi and connecting again.

@pathikb @ratelgogo You might workaround by using the [Atlantis framework](https://github.com/ProxymanApp/atlantis) (develop by Proxyman team), to capture HTTPS traffic.

**Comment by yuriibabii** [comment]
Was able to fix this after installing Proxyman app on the phone, installing its certificate, enabling VPN, verified it's working. Then turn off the VPN, start setuping proxy for desktop again. Now the certificates should be downloaded and able to install of the device. After that everything worked as expected. Doesn't matter if it's Proxyman desktop app, or Charles.

**Comment by rounak** [comment]
Forgetting and rejoining the network worked!

**Comment by dnesterov** [comment]
iOS 16.6 - Still not working :(

**Comment by NghiaTranUIT** [comment]
@dnesterov try to forget the Wifi and reconnect again. Set the proxy, then Save it TWICE. For some reason, the iOS doesn't save the proxy setting.

**Comment by dnesterov** [comment]
> commented

Thank you, it is working after forgetting Wi-Fi!

**Comment by IDEKENTERPRISES** [comment]
Currently not working on iOS 16.7.2, tried forgetting network and restarting iPhone to no avail; traffic continues to ignore proxy setting.

**Comment by Dast1Woop** [comment]
iOS17,still not working

**Comment by ouzihao** [comment]
> Currently not working on iOS 16.7.2, tried forgetting network and restarting iPhone to no avail; traffic continues to ignore proxy setting.
iOS 16.7.1 have the same problem too.

**Comment by castrojr913** [comment]
> Currently not working on iOS 16.7.2, tried forgetting network and restarting iPhone to no avail; traffic continues to ignore proxy setting.

I was facing the same issue with my app on iOS 17, but I found the HTTP client (Alamofire) blocked the interception. For instance, using `Alamofire.SessionManager.default`. It even displayed the message "The Certificate for this server is invalid", canceling the request. Looking into a developer's answer on [StackOverflow](https://stackoverflow.com/a/44169463) to disable this behavior, I implemented a custom ServerTrustPolicyManager, to initialize the session manager for each request:

```
open class MyServerTrustPolicyManager: ServerTrustPolicyManager {
        open override func serverTrustPolicy(forHost host: String) -> ServerTrustPolicy? {
            return ServerTrustPolicy.disableEvaluation
        }
}

let sessionManager = SessionManager(delegate:SessionDelegate(), serverTrustPolicyManager:MyServerTrustPolicyManager(policies: [:]))
```

In summary, in my case, ProxyMan was working properly, Alamofire default configuration was the issue instead.

**Comment by NghiaTranUIT** [comment]
@castrojr913 It's not really related to this issue, which is an Apple Bug.

Some iPhones didn't use HTTP Proxy Config in the WiFi Setting. 

So far, here are solutions from Proxyman users.
- Forget the Wi-Fi
- Restart the iPhone
- Double-check the IP & Port to match with Proxyman
- Open [Firewall setting](https://docs.proxyman.io/troubleshooting/my-ios-devices-couldnt-connect-to-proxyman-via-proxy#4.-firewall-settings-macos) to accept all incoming traffic (iPhone)
- and update to the latest iOS 17

**Comment by tutuerhao** [comment]
![IMG_4147](https://github.com/ProxymanApp/Proxyman/assets/29938085/df7fcdc3-cabf-4ebd-b194-b36240395b50)
重置网络可能会有帮助

**Comment by tutuerhao** [comment]
You can use proxy tools such as Shadowrocket, Surge, and Loon to add strategies for packet capture.

**Comment by NghiaTranUIT** [comment]
@GioPalusa but your iPhone never uses the HTTP Proxy, so Proxyman macOS can't see any request from your iPhone.

This issue is fixed by Apple on iOS 17. You can see some solutions at https://docs.proxyman.io/troubleshooting/ios-16-devices-issues

**Comment by JauYang** [comment]
I have tried all the solutions in this issues. The only thing that is feasible is to forget the network, restart the phone, and then re-execute all the actions, but this will not be solved for a long time, and the phone will fail after a few minutes of resting;

iOS 17.3.1, the problem still cannot be dealt with effectively, and it is too troublesome to restart the phone.
f**k apple

**Comment by NghiaTranUIT** [comment]
@JauYang can you confirm that there is no active VPN when you set the proxy setting? Some users had active VPN and didn't realize to turn it off.

**Comment by 22Round** [comment]
As of iOS 17.4.1, the issue is till there :((

**Comment by Sjahriyar** [comment]
Forget Wifi doesn't work anymore on iOS 17.4.1
I have even tried Atlantis framework, that also doesn't work unfortunately

**Comment by NghiaTranUIT** [comment]
@Sjahriyar can you try each step in the Troubleshooting: https://docs.proxyman.io/troubleshooting/my-ios-devices-couldnt-connect-to-proxyman-via-proxy

Maybe it's from your Firewall, or VPN app, ...

**Comment by GioPalusa** [comment]
I did the download on a Mac and airdrop the certificate to a new iPhone yesterday and it worked like a charm. I capture all SSL traffic with no hassle.
So go to the http://proxy.man/ssl from your computer and download the cert from there and airdrop it to your phone, and then install it and trust it. Then it works just as it should.

**Comment by NghiaTranUIT** [comment]
@GioPalusa if you can capture from your iPhone, it means the Proxy on your iPhone is working fine. You can access http://proxy.man/ssl to download it from your phone too.

If you can't, I guess it's HTTPS because Safari might use it by default.

**Comment by Bert-Harvey-kr** [comment]
I was able to transfer the cert from my mac to my physical device running iOS 17 and it displays as installed and trusted, but i'm still not capturing anything through the proxy is set on the phone and traffic is getting through without issue. What am I failing to do here?

**Comment by NghiaTranUIT** [comment]
@Bert-Harvey-kr it means your iPhone has a bug when setting the HTTP Proxy on Wifi to Proxyman. You can confirm that it's an Apple bug by:
1. Open Setting app
2. Wifi -> Your wifi -> Proxy 
3. Use any invalid IP and Port, for example: 192.1.1.1 and port 8000 -> Save
4. Open Safari -> www.google.com -> If you can access the Internet -> It's a bug from Apple.

It's fixed on iOS 17.3 or later. You can find the doc at https://docs.proxyman.io/troubleshooting/ios-16-devices-issues

You should try to forget the network, restart your iPhone, or upgrade to the latest iOS 👍

**Comment by evancauwenberg** [comment]
Still an issue on iOS 17.5 developer beta, did all the possible trouble shooting :(

**Comment by Panajev** [comment]
Is this an issue with iOS 18 too?

**Comment by NghiaTranUIT** [comment]
from what I know, iOS 18 works totally fine 👍 the bug is fixed.

**Comment by Panajev** [comment]
Thank you @NghiaTranUIT would it be possible to update the official website docs to reference that iOS 18 is free from this bug please?

Also, I would guess this bug could be closed right? @evancauwenberg @Esquilli have you found that it is indeed solved for you on iOS 18?

**Comment by NghiaTranUIT** [comment]
It's updated on the doc 👍 

If it doesn't work, try to forget your Wi-Fi and try again. I works on my iPad and iPhone, both iOS 18 👍 

![Screenshot 2024-10-11 at 9 17 31 PM](https://github.com/user-attachments/assets/e053ba06-3d71-46ae-827a-8d48357dd5c8)

**Comment by NghiaTranUIT** [comment]
@barayuda can you open Proxyman Mac -> Settings -> General Tab -> And share with me the name of your certificate? 

Do you see the certificate in  Setting app > General > About > Certificate Trust Settings > Switch ON on Proxyman CA. ?

**Comment by NghiaTranUIT** [comment]
Glad it fixed 👍

**Comment by maciejpigulski** [comment]
I still had issue on iOS 18 and after much trial and errors I have discovered what was wrong with my setup.

When developing I have an iPhone connected via USB cable to my MacBook Pro. Given that on macOS I had `Internet Sharing` enabled in `Settings->General->Sharing->iPhone USB` then WiFi connection got ignored. iPhone starts using the internet via USB from my MBP.

After I disconnect the cable, then a miracle happens and WiFi starts to be utilized and Proxy settings start to work so everything starts working through Proxyman.

Solution for me to both have the USB cable used for app development and installation was to disable `Internet Sharing` completely so iPhone started using the WiFi again.

Maybe this will help someone.
